### PR TITLE
Version 4.12.0

### PR DESCRIPTION
**Pull request by danfinlay**
Release Candidate.

- Merged master into develop to get latest version notes & changes (includes updated contract-metadata).
- Fixed merge conflicts.
- Bumped version & changelog using `npm run version:bump minor`.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -10288,10 +10288,25 @@
       "resolved": "https://registry.npmjs.org/eth-sig-util/-/eth-sig-util-2.0.2.tgz",
       "integrity": "sha512-tB6E8jf/aZQ943bo3+iojl8xRe3Jzcl+9OT6v8K7kWis6PdIX19SB2vYvN849cB9G9m/XLjYFK381SgdbsnpTA==",
       "requires": {
+<<<<<<< HEAD
```
Comment (whymarrh) [inline package-lock.json]: We'll want to remove these merge markers

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (danfinlay) [reply package-lock.json]: whoooops, missed that one :(

**Review by brunobar79 — DISMISSED** [review]
Looks good to me! Let's 🚀 !

File: package-lock.json (hunk 1 above)
Comment (danfinlay) [reply package-lock.json]: Fixed. Running `npm i` fixes package-lock conflicts.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 ## Current Develop Branch
 
+## 4.12.0 Tuesday September 25 2018
```
Comment (whymarrh) [inline CHANGELOG.md]: We should also update this with today's date

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (danfinlay) [reply CHANGELOG.md]: Thank you for the eagle eyes!

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,18 @@
 
 ## Current Develop Branch
 
+## 4.12.0 Tuesday September 27 2018
```
Comment (whymarrh) [inline CHANGELOG.md]: Today is Thursday 😄

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (danfinlay) [reply CHANGELOG.md]: Apparently my `version:bump` script doesn't help me out when there's no changelog yet, so I copy-pasted and now pay for it.

**Review by whymarrh — APPROVED** [review]
LGTM

**Pull request merged**